include/sys/__assert.h

```diff
@@ -126,4 +130,8 @@ void assert_post_action(const char *file, unsigned int line);
 #define __ASSERT_NO_MSG(test) { }
 #endif
 
+#ifdef __cplusplus
```

Comment:
This will have to move to be at the same level as the moved `extern "C"`.

Reply:
@pabigot  better ?